src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -1369,7 +1369,7 @@ static inline int MPIDI_OFI_application_hints(int rank)
         MPIR_ERR_SETANDJUMP1(mpi_errno, MPI_ERR_OTHER, "**pmi_get_size",
                              "**pmi_get_size %d", mpi_errno);
     }
-    if (world_size > (1UL << MPIDI_OFI_MAX_RANK_BITS)) {
+    if ((size_t)world_size > ((size_t)1 << MPIDI_OFI_MAX_RANK_BITS)) {
```

Comment:
my vision was size_t is more flexible data type (in fact here are two sizes are compared). bu ok, uint64_t is ok too. updated